### PR TITLE
changed default ciphers to modern ones

### LINKED ISSUE #8581: TLS connections should disallow handshakes which use small DH groups
See for example:
- https://dh480.badssl.com/
- https://dh512.badssl.com/

which should be blocked. (https://dh1024.badssl.com/ should also be blocked eventually IMO, but is not yet in many major browsers)

### PR DESCRIPTION
**Pull request by avadacatavra**
I changed the default ciphers to the modern set (ref: https://wiki.mozilla.org/Security/Server_Side_TLS), but added an option to use the intermediate set for web compatibility

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #8581 (github issue number if applicable).

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16535)

### CODE REVIEW

**Comment by jdm** [comment]
This looks like a reasonable way to experiment with not supporting ciphers that might still be common.
@bors-servo: r+

**Comment by jdm** [comment]
Whoops. Apparently the certificates used for the HTTPS server for WPT rely on some ciphers that are now ignored by default.

**Comment by avadacatavra** [comment]
@jdm do we want new certs or to just add those into the default?

**Comment by jdm** [comment]
@jgraham Do you think there's any objection to the cert-generation change in web-platform-tests/tools?

**Comment by jgraham** [comment]
Do we expect it to change our compatibility story? Should these certs work with latest Firefox, Chrome, Edge and Safari releases? If they will it seems fine to me, otherwise not. Please remember to upstream the change.

**Comment by jgraham** [comment]
(oh it's in `wpt/tools` so can upstream automatically now)

**Comment by avadacatavra** [comment]
@jgraham according to [this](https://wiki.mozilla.org/Security/Server_Side_TLS),I think it should be compatible with latest releases. I've changed the certs to use ECDSA with curve secp521r1.

**Comment by jgraham** [comment]
OK, then I'm happy with this change.

**Comment by jdm** [comment]
@bors-servo: r-
Actually, I'll email public-test-infrastructure first just in case.

File: components/net/connector.rs [hunk 1]
```diff
@@ -51,11 +52,16 @@ impl NetworkConnector for HttpsConnector {
 pub type Connector = HttpsConnector;
 
 pub fn create_ssl_client(ca_file: &PathBuf) -> OpensslClient {
+    let opts = opts::get();
+    let cipher_list = match opts.cipher_compatibility {
```
Comment (emilio) [inline components/net/connector.rs]:
```rust
let cipher_list = if opts::get().cipher_compatibilty {
    INTERMEDIATE_CIPHERS
} else {
    MODERN_CIPHERS
}
```

may be slightly more readable?

Comment (avadacatavra) [inline components/net/connector.rs]: @emilio fine with me

**Comment by jdm** [comment]
Others are favourable; ekr says "Yeah, don't use P521. Use P256 or P384.".

File: tests/wpt/web-platform-tests/tools/sslutils/openssl.py [hunk 2]
```diff
@@ -402,13 +406,18 @@ def _generate_host_cert(self, hosts):
         self.logger.info("Generating new host cert")
 
         with self._config_openssl(hosts) as openssl:
+            openssl("ecparam",
+                    "-name", "secp521r1",
```
Comment (avadacatavra) [inline tests/wpt/web-platform-tests/tools/sslutils/openssl.py]: ...hmph will fix

**Comment by jdm** [comment]
This does not appear to work. All the https tests are timing out.

**Comment by avadacatavra** [comment]
I'll investigate

**Comment by metajack** [comment]
@avadacatavra what's the next step here?

**Comment by jdm** [comment]
Closing due to inactivity.

**Pull request closed without merging**